tests/unittests/test_expression_attributes_log_based.py

```diff
@@ -0,0 +1,150 @@
+import unittest
```

Comment:
Unittest cases of prepare_projection must be separate from the sync method tests

Test expression attribute with empty string passed in the expression field
Test expression attribute with empty string passed in projection field.

These 2 must be covered separately after mocking the prepare_projection function.

Reply:
We have tested the sync() method because there are scenarios where there can be multiple projections, and in the code, the `prepare_projection` is looped over the list of projections.
Added the testcases to verify the `prepare_projection` is called when the expressions passed in the catalog, and when not passed.

Reply:
Make sense